bundles/org.openhab.binding.mercedesme/src/main/java/org/openhab/binding/mercedesme/internal/server/CallbackServer.java

```diff
@@ -0,0 +1,190 @@
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.mercedesme.internal.server;
+
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.server.Connector;
+import org.eclipse.jetty.server.Server;
+import org.eclipse.jetty.server.ServerConnector;
+import org.eclipse.jetty.servlet.ServletHandler;
+import org.openhab.binding.mercedesme.internal.AccountConfiguration;
+import org.openhab.binding.mercedesme.internal.Constants;
+import org.openhab.core.auth.client.oauth2.AccessTokenRefreshListener;
+import org.openhab.core.auth.client.oauth2.AccessTokenResponse;
+import org.openhab.core.auth.client.oauth2.OAuthClientService;
+import org.openhab.core.auth.client.oauth2.OAuthException;
+import org.openhab.core.auth.client.oauth2.OAuthFactory;
+import org.openhab.core.auth.client.oauth2.OAuthResponseException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link CallbackServer} class defines an HTTP Server for authentication callbacks
+ *
+ * @author Bernd Weymann - Initial contribution
+ */
+@NonNullByDefault
+public class CallbackServer {
+    private static final Logger LOGGER = LoggerFactory.getLogger(CallbackServer.class);
+    private static final Map<Integer, OAuthClientService> AUTH_MAP = new HashMap<Integer, OAuthClientService>();
+    private static final Map<Integer, CallbackServer> SERVER_MAP = new HashMap<Integer, CallbackServer>();
+
+    private Optional<Server> server = Optional.empty();
+    private Optional<AccessTokenResponse> token = Optional.empty();
+    private AccessTokenRefreshListener listener;
+    private AccountConfiguration config;
+    private OAuthClientService oacs;
+    private String callbackUrl;
+
+    public CallbackServer(AccessTokenRefreshListener l, HttpClient hc, OAuthFactory oAuthFactory,
+            AccountConfiguration config, String callbackUrl) {
+        oacs = oAuthFactory.createOAuthClientService(Constants.OAUTH_CLIENT_NAME + config.callbackPort,
+                Constants.MB_TOKEN_URL, Constants.MB_AUTH_URL, config.clientId, config.clientSecret, config.getScope(),
+                false);
+        AUTH_MAP.put(Integer.valueOf(config.callbackPort), oacs);
+        SERVER_MAP.put(Integer.valueOf(config.callbackPort), this);
+        listener = l;
+        this.config = config;
+        this.callbackUrl = callbackUrl;
+    }
+
+    public String getAuthorizationUrl() {
+        try {
+            return oacs.getAuthorizationUrl(callbackUrl, null, null);
+        } catch (OAuthException e) {
+            LOGGER.error("Error creating Authorization URL {}", e.getMessage());
```

Comment:
In this situation, will the binding stop working completely? Otherwise log level should be reduced - for reference: https://www.openhab.org/docs/developer/guidelines.html#f-logging

Reply:
Reduced to warning - binding won't stop completely but manual authorization will fail which is crucial for the initial setup